upstairs/src/client.rs

```diff
@@ -380,16 +375,17 @@ impl DownstairsClient {
 
         // Update our bytes-in-flight counter
         if was_running && !is_running {
-            self.bytes_outstanding = self
-                .bytes_outstanding
-                .checked_sub(job.work.job_bytes())
-                .unwrap();
-            self.write_bytes_outstanding.decrement(job, self.client_id);
+            // Because the job is no longer running, it shouldn't count for
+            // backpressure.  Remove the backpressure guard for this client,
+            // which decrements backpressure counters on drop.
+            job.backpressure_guard.take(&self.client_id);
         } else if is_running && !was_running {
             // This should only happen if a job is replayed, but that still
             // counts!
-            self.bytes_outstanding += job.work.job_bytes();
-            self.write_bytes_outstanding.increment(job, self.client_id);
+            job.backpressure_guard.insert(
```

Comment:
I realize this is not a change in when we adjust backpressure, but how are we not counting replay IOs twice here?  Once for the original IO and a 2nd time on the replay?  I think I've missed a decrement step somewhere.

Reply:
The decrement happens automatically when we replace the `BackpressureGuard` in the `ClientMap`, because the old value is dropped (and decrements on drop).